Add option to toggle visibility of Position gizmos in 2D editor, organize existing options

J-N-Witch opened the pull request:
This change lets you hide Position gizmos on 2D nodes
allowing you to make clean screenshots in the editor without crosses everywhere, among other things.

In order to avoid clutter in 2D > View menu, "Show Group And Lock Icons" and "Show Gizmos"
were moved to "Gizmos" submenu along with the newly added option.

Before:

![4](https://user-images.githubusercontent.com/127793256/225105554-4b6f5230-c216-4913-839e-169112dee29d.png)

After:

https://user-images.githubusercontent.com/127793256/225758780-4f5ce8be-785f-499a-8c9c-6ce9dac04b5c.mp4

Related proposal: https://github.com/godotengine/godot-proposals/issues/6344

show_edit_locks (which actually referred to showing locks _and groups_)
was renamed to show_lock_gizmos to be consistent with the existing show_transformation_gizmos;
and was decoupled from group gizmos.

I would like have this in 3.x as well.

<!--
Please target the `master` branch in priority.
PRs can target `3.x` if the same change was done in `master`, or is not relevant there.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.
You can mention in the description if the change is compatible with `3.x`.
-->

Drkie commented:
@YuriSizov any reason this got moved from 4.1? It doesn't seem like it will make it to 4.2 either.

YuriSizov commented:
@Drkie Nobody reviewed it thus far, so it can't be considered for the next release. Indeed, it won't make the cut for 4.2 either at this point, because we're entering feature freeze.

KoBeWi commented on editor/plugins/canvas_item_editor_plugin.cpp:
```diff
@@ -3575,7 +3575,7 @@ void CanvasItemEditor::_draw_invisible_nodes_positions(Node *p_node, const Trans
 	}
 
-	if (ci && !ci->_edit_use_rect() && (!editor_selection->is_selected(ci) || _is_node_locked(ci))) {
+	if (ci && !ci->_edit_use_rect() && (!editor_selection->is_selected(ci) || _is_node_locked(ci)) && show_position_gizmos) {
```
```suggestion
	if (show_position_gizmos && ci && !ci->_edit_use_rect() && (!editor_selection->is_selected(ci) || _is_node_locked(ci))) {
```
Cheapest condition first.

J-N-Witch replied:
Do you want to have the other 2 checks reordered as well?

Line 3714 and 3720
`if (p_node->has_meta("_edit_lock_") && show_lock_gizmos) {`
to
`if (show_lock_gizmos && p_node->has_meta("_edit_lock_")) {`

AThousandShips replied:
Yes

KoBeWi commented:
IMO a single-element separated section is a bit awkward:
![image](https://github.com/godotengine/godot/assets/2223172/49010f5f-32fa-4319-b90c-643dba943595)
Maybe the Gizmos could be part of the first section. You could move the Grid to the bottom of the section, so two menus would be next to each other. Also Clear Guides is at odd spot, it could be near the checkboxes too (though not sure).

AThousandShips commented on editor/plugins/canvas_item_editor_plugin.cpp:
```diff
@@ -5275,8 +5301,9 @@ CanvasItemEditor::CanvasItemEditor() {
-	p->add_check_shortcut(ED_SHORTCUT("canvas_item_editor/show_transformation_gizmos", TTR("Show Transformation Gizmos")), SHOW_TRANSFORMATION_GIZMOS);
+	p->add_separator();
+
+	p->add_submenu_item(TTR("Gizmos"), "GizmosMenu");
```
This should be after the menu has been added IMO, so either move this down or the declaration of the menu up

J-N-Witch commented:
> IMO a single-element separated section is a bit awkward

This is exactly how the existing Gizmos menu in 3D looks like.

![3d-gizmos](https://github.com/godotengine/godot/assets/127793256/a18515e0-b487-4a73-9dd0-fa98ef7785ee)

The pattern is also repeated in many other places. See "Scene > Export As...", "Project > Version Control" etc.

Anyways, does this look better or should we keep the current layout?

![2d-gizmos-updated](https://github.com/godotengine/godot/assets/127793256/bcd193c5-34ca-425c-b3e3-15181369d571)


> Also Clear Guides is at odd spot, it could be near the checkboxes too (though not sure).

I think messing with Clear Guides is out of scope for this change.

J-N-Witch committed:
Introduce a "Gizmos" submenu for toggling visibility of gizmos in 2D editor
instead of standalone checkboxes like "Show Group And Lock Gizmos".

Add option to toggle visibility of Position gizmos.

J-N-Witch commented:
Any thoughts on my comment above @AThousandShips @KoBeWi ?

I have rebased the change on the latest master and implemented your suggestions, let me know if there's anything else I can do.

AThousandShips reviewed (approved):
Haven't tested but code looks good

akien-mga commented:
Thanks! And congrats for your first merged Godot contribution :tada:
